### ISSUE TITLE
stopAfterElapsedMinutes behavior

### ISSUE THREAD

**Issue by kareldebedts** [transistorsoft/flutter_background_geolocation#384]
## Your Environment
* Plugin version: 1.10.0
* Platform: iOS
* OS version: IOS 13
* Device manufacturer / model: iPhone 11

## Expected Behavior
I would expect that if I call: bg.BackgroundGeolocation.start(); x minutes later bg.BackgroundGeolocation.stop(); will be called.

## Actual Behavior
Tracking doesn't always stop after x minutes. It's very random.

Do you have more info about the behavior?

When does it start "counting": when in stationary state or moving state?
When does it count: always or only when in moving state?
When does it stop: only if the app is opened or also in background?

I had very mixed results with testing by sitting for x minutes and walking for x minutes.

Thanks!

**Comment by christocracy** [comment]
It starts counting the first time the device is detected to be *moving*, where `onMotionChange` event fires with `isMoving == true‘.

**Comment by kareldebedts** [comment]
Okay thanks for the quick response.

So if I run bg.BackgroundGeolocation.changePace(true) after bg.BackgroundGeolocation.start(); it will count whatever happens? (even if the app closes or if it goes in stationary mode?) Cuz it's important that it really stops x minutes later after bg.BackgroundGeolocation.start();

**Comment by christocracy** [comment]
> even if the app closes or if it goes in stationary mode?

No.  Once your app suspends in the background in stationary state, there’s no code running.  The plugin can only check the next time your app awakens.

**Comment by kareldebedts** [comment]
Ok so let’s say we’re tracking 60 minutes.

If I start 5 minutes in moving state. Then 50 in stationary. And then 5 moving. Will it stop or will it only count The minutes in moving State to achieve 60 ?

**Comment by christocracy** [comment]
Why don’t you try it and see?

**Comment by kareldebedts** [comment]
Yes I will. But it would save me some time if I knew what to expect. So is the expected behavior that it will start counting as soon as you are in moving mode and will keep counting even if you are in stationary? I would really appreciate your  help. Thanks !

**Comment by kareldebedts** [comment]
I tried it.

I've set stopAfterElapsed minutes to 30

started tracking with:

bg.BackgroundGeolocation.start();
bg.BackgroundGeolocation.changePace(true)

After 30 minutes, it still tracks, because I've been in stationary mode for 20 minutes.
So I think it only "counts" when in moving mode to get to the 30 minutes.


Are there other options for having a timer that always count?
I thought of running BackgroundGeolocation.onLocation function & checking with shared preferences how many minutes are left. Don't know if it's possible to run onLocation in the background.

Thanks!
